src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -123,15 +116,13 @@ private function findClasses($namespace, $pattern, $excludePattern)
             if (!preg_match($extRegexp, $path, $m) || !$info->isReadable()) {
                 continue;
             }
-            $class = $namespace.ltrim(str_replace('/', '\\', substr($path, $prefixLen, -strlen($m[0]))), '\\');
 
-            if (!preg_match('/^[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*+(?:\\\\[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*+)*+$/', $class)) {
+            if (!$class = $this->getClassFromFile($path)) {
+                // no class found in file
                 continue;
             }
-            // check to make sure the expected class exists
-            if (!$r = $this->container->getReflectionClass($class)) {
-                throw new InvalidArgumentException(sprintf('Expected to find class "%s" in file "%s" while importing services from resource "%s", but it was not found! Check the namespace prefix used with the resource.', $class, $path, $pattern));
-            }
+
+            $r = $this->container->getReflectionClass($class);
```

Comment:
I think this may still fail if the parent class is missing. Can you confirm @nicolas-grekas ?

Reply:
not anymore
but anyway, this implementation is going to change completly (see my other comments)